datasets/saudinewsnet/saudinewsnet.py

```diff
@@ -0,0 +1,102 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The dataset contains a set of 2,972 Arabic newspaper articles alongwith metadata, extracted from various online Saudi newspapers."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+
+import datasets
+
+
+_CITATION = """\
+@misc{hagrima2015,
+author = "M. Alhagri",
+title = "Saudi Newspapers Arabic Corpus (SaudiNewsNet)",
+year = 2015,
+url = "http://github.com/ParallelMazen/SaudiNewsNet"
+}
+"""
+
+
+_DESCRIPTION = """The dataset contains a set of 2,972 Arabic newspaper articles alongwith metadata, \
+extracted from various online Saudi newspapers and written in MSA."""
+
+
+_HOMEPAGE = "https://github.com/parallelfold/SaudiNewsNet"
+
+
+_LICENSE = "Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License."
+
+
+_URL = "https://github.com/parallelfold/SaudiNewsNet/raw/master/dataset/2015-07-21.zip"
+
+
+class SaudiNewsNet(datasets.GeneratorBasedBuilder):
+    """a set of 2972 Arabic newspaper articles along with metadata"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "source": datasets.Value(
+                        "string"
+                    ),  # A string identifief of the newspaper from which the article was extracted.
+                    "url": datasets.Value("string"),  # The full URL from which the article was extracted.
+                    "date_extracted": datasets.Value(
+                        "string"
+                    ),  # The timestamp of the date on which the article was extracted.
+                    "title": datasets.Value("string"),  # The title of the article. Can be empty.
+                    "author": datasets.Value("string"),  # The author of the article. Can be empty.
+                    "content": datasets.Value("string"),  # The content of the article.
+                }
+            ),
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+            supervised_keys=None,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        datadir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(datadir, "2015-07-21.json"), "split": "train"},
```

Comment:
Can you re-add all the other batches as well ?

Sorry I didn't notice that in my previous review

Reply:
@lhoestq  No worries, I'm learning A LOT. I'm the one who's sorry to bother you with my newbie mistakes. 

I need your advice, I failed before in adding the batches because i couldn't figure out how to read from all the links. There are 18 links for each batch of articles.

If I create a list of filepaths, would the dl manager be able to read and extract the data? or are there better ways to handle multiple links for download?


I would've tested the idea now but I just went out and won't be back in an hour.

Reply:
The dl_manager.download_and_extract method can take a list of URLs as input. The output will be a list of paths to the extracted data.

Moreover a nice addition is that all the files are going to be downloaded in parallel, which is not the case if you call dl_manager.download_and_extract multiple times in a row.

Reply:
very cool. I'll fix it shortly. I'm grateful for your feedback

Reply:
@lhoestq added the remaining 17 batches and modified the readme.md to reflect that + resolved the camel case comment